lib/p2p/node/supervisor.ex

```diff
@@ -91,14 +91,39 @@ defmodule Elixium.Node.Supervisor do
           body
           |> Jason.decode!()
           |> Enum.map(&peerstring_to_tuple/1)
-          |> Enum.filter(fn {_, port} -> port != nil end)
+          |> Enum.filter(fn {peer, port} -> port != nil && peer !== fetch_public_ip["ip"] && peer !== fetch_local_ip end)
```

Comment:
Here we're fetching our public ip for every peer in the list. Let's fetch our IP beforehand and just check against it here instead

Reply:
Also -- we shouldnt just filter out our own IP, we should only filter it out if it matches both the port we're running on and our IP. This will allow us to connect to our other projects (miner/wallet) locally, otherwise we'll only be able to run one of the projects at a time, regardless of port.

Reply:
fixed and updated